### PR TITLE
Error handling on types failures

### PR DESCRIPTION
**Pull request by PelegLi**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

## Description

several minor fixes for type mismatches referencing not initialized variables

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code

### CODE REVIEW

**Review by ChanochShayner — APPROVED** [review]

File: checkov/helm/runner.py [hunk 1]
```diff
@@ -155,10 +155,16 @@ def _convert_chart_to_k8s(self, chart_item):
...
         chart_version = chart_meta.get('version', chart_meta.get('Version'))
-        if target_dir.endswith('/'):
-            target_dir = target_dir[:-1]
-        if target_dir.endswith(chart_name):
-            target_dir = target_dir[:-len(chart_name)]
+        try:
+            if target_dir.endswith('/'):
+                target_dir = target_dir[:-1]
+            if target_dir.endswith(chart_name):
+                target_dir = target_dir[:-len(chart_name)]
```
Comment (YaaraVerner) [inline checkov/helm/runner.py:162]: what was the error here? the length of the string must be enough (the `endswith` function should verify it)

File: checkov/terraform/checks/resource/gcp/CloudArmorWAFACLCVE202144228.py [hunk 2]
```diff
@@ -24,14 +24,17 @@ def scan_resource_conf(self, conf: Dict[str, List[Any]]) -> CheckResult:
...
+                    expr = match[0].get("expr")
+                    if expr and expr[0].get("expression") == ["evaluatePreconfiguredExpr('cve-canary')"]:
+                        if rule.get("action") == ["allow"]:
+                            return CheckResult.FAILED
+                        if rule.get("preview") == [True]:
+                            return CheckResult.FAILED
+
+                        return CheckResult.PASSED
+                except Exception:
+                    return CheckResult.UNKNOWN
```
Comment (YaaraVerner) [inline checkov/terraform/checks/resource/gcp/CloudArmorWAFACLCVE202144228.py:37]: can you check the specific edge case? and add test for it :)

**Review by YaaraVerner — COMMENTED** [review]
had some comments :)

**Review by achiar99 — APPROVED** [review]

File: checkov/helm/runner.py (hunk 1 above)
Comment (PelegLi) [reply checkov/helm/runner.py:162]: TypeError: endswith first arg must be str or a tuple of str, not NoneType

Comment (gruebel) [reply checkov/helm/runner.py:162]: if `target_dir` is  `None`, then a few lines higher `target_dir.replace("//", "/")` should also fail.
probably just do a `None` check after line 154.
```python
target_dir = chart_dir.replace(self.root_folder, f'{self.target_folder_path}/')
if target_dir is None:
    raise ...
```

File: checkov/kustomize/runner.py [hunk 3]
```diff
@@ -229,7 +229,7 @@ def check_system_deps(self):
                         self.templateRendererCommand = self.kubectl_command
                         return None
             
-            except Exception:
+            except Exception as e:
                 logging.debug(f"An error occured testing the {self.kubectl_command} command: {e}")
```
Comment (gruebel) [inline checkov/kustomize/runner.py:233]:
```suggestion
            except Exception:
                logging.debug(f"An error occured testing the {self.kubectl_command} command", exc_info=True)
```
then we have the complete stack trace

File: checkov/kustomize/runner.py [hunk 4]
```diff
@@ -248,7 +248,7 @@ def check_system_deps(self):
                 else:
                     return self.check_type
 
-            except Exception:
+            except Exception as e:
```
Comment (gruebel) [inline checkov/kustomize/runner.py:251]: same here

File: checkov/terraform/checks/resource/gcp/CloudArmorWAFACLCVE202144228.py (hunk 2 above)
Comment (gruebel) [reply checkov/terraform/checks/resource/gcp/CloudArmorWAFACLCVE202144228.py:37]: that would be great, I'm also curious what exactly is failing here and why.

File: checkov/helm/runner.py (hunk 1 above)
Comment (PelegLi) [reply checkov/helm/runner.py:162]: @gruebel but it didn't fail on a few lines higher, it fail on `if target_dir.endswith(chart_name):`

File: checkov/terraform/checks/resource/gcp/CloudArmorWAFACLCVE202144228.py (hunk 2 above)
Comment (PelegLi) [reply checkov/terraform/checks/resource/gcp/CloudArmorWAFACLCVE202144228.py:37]: the case was that `rule` is a list, so the `try-except` catches this case of unexpected type and also unexpected `expr` type, in case there will be

File: checkov/helm/runner.py (hunk 1 above)
Comment (gruebel) [reply checkov/helm/runner.py:162]: ah got it, the problem is the `chart_name`, it could be `None`. This should work on line 156
```python
chart_name = chart_meta.get('name', chart_meta.get('Name'))
if chart_name is None:
    raise ...
```
